src/DDTrace/Integrations/Memcached/MemcachedIntegration.php

```diff
@@ -60,70 +60,123 @@ public static function load()
 
         // bool Memcached::add ( string $key , mixed $value [, int $expiration ] )
         dd_trace('Memcached', 'add', function () {
+            if (GlobalTracer::get()->limited()) {
+                return dd_trace_forward_call();
```

Comment:
Not sure if `dd_trace_forward_call()` would work when called from `traceCommand` or `traceCommandByKey`. If yes then we may want to move this check over there? If not we should create a task to make sure that this can be done in the future as it seems pretty useful. WDYT?

Reply:
Ok probably

> I've considered changing existing api to return null spans

seems the answer to my question 😄

Reply:
Actually when I first implemented it it was called like that. But there is a limitation of dd_trace_forward_call() that it only works from within the closure.

Overall I think we could change the traceCommandByKey API it a bit to make that possible.
For now more pragmatic way was to just copy paste this tidbit.  In general however I would see us move away from using `dd_trace` directly.

Reply:
How does this wrk then? https://github.com/DataDog/dd-trace-php/pull/417/files#diff-5ed8e1b32015797d9cf19b5ce8f354b9R53

Reply:
How does this work then? https://github.com/DataDog/dd-trace-php/pull/417/files#diff-5ed8e1b32015797d9cf19b5ce8f354b9R53

Reply:
@labbati because of 
https://github.com/DataDog/dd-trace-php/pull/417/files#diff-5ed8e1b32015797d9cf19b5ce8f354b9R43 :D (i.e. it runs directly within dd_trace closure)

Reply:
I meant the underlying root cause......and the underlying root cause is.....ME :palmface:

Reply:
:) Overall I think we could make dd_trace_forward_call call work from whatever context, but its a C level change, and I'm worried about unexpected consequences. 
And I liked our hands to be tied a little here, due to performance, we shouldn't do a lot of calls/allocations/objects from within the closure. 

Also it makes stacktraces make more sense if everything is happening within dd_trace.

I think following existing examples of `tracePublicMethod` and some functional programming concepts we could make nice idempotent functions passed to dd_trace flat and uber-fast.

Reply:
👍 totally agree.